apps/keyboard/test/build/integration/keyboard_test.js

```diff
@@ -100,10 +100,18 @@ suite('Keyboard layouts building tests', function() {
 
           return path.basename(filename, '.js');
         });
-
-      var layouts = layoutIds.map(function(layout) {
-        return 'js/layouts/' + layout + '.js';
-      });
+      
+     var layoutIds1 = [];
+     layoutIds1 = layoutIds.filter(function(layout) {
+     	if(layout)
+     		return layout;
+     });
+     
+     layoutIds = layoutIds1;
```

Comment:
You can chain these arrays with `var layouts = layoutIds.filter(...).map(...);`

Where the `filter()` function will simply return the array (the one you named `layoutIds1`) implicitly for you to call (`map()`), and `layouts` will be the value returning from `map()`.